src/Path.php

```diff
@@ -189,7 +189,7 @@ public static function check($path)
 	 */
 	public static function clean($path, $ds = DIRECTORY_SEPARATOR)
 	{
-		if (!is_string($path))
+		if (!is_string($path) && !empty($path))
```

Comment:
That's what's getting me confused.  It almost seems like it's saying "if this is a string and it has characters throw an exception".